lib/web_ui/test/engine/text_editing_test.dart

```diff
@@ -532,6 +532,12 @@ Future<void> testMain() async {
               Matrix4.translationValues(10.0, 20.0, 30.0).storage.toList());
       sendFrameworkMessage(codec.encodeMethodCall(setSizeAndTransform));
 
+      // This timer exists because Desktop Safari element is focused after
+      // zero-duration timer to prevent autofill dialog
+      if(isSafari){
+        await Future<void>.delayed(Duration.zero);
+      }
```

Comment:
This block (comment + code) is being repeated a lot. Should we maybe extract this into a function?
```dart
/// On Desktop Safari, the editing element is focused after zero-duration timer
/// to prevent autofill dialog ...
Future<void> waitForDesktopSafariFocus() async {
  if (editingStrategy is SafariDesktopTextEditingStrategy) {
    await Future<void>.delayed(Duration.zero);
  }
}
```

```suggestion
      await waitForDesktopSafariFocus();
```